src/PAModelTests/ErrorTests.cs

```diff
@@ -55,5 +58,148 @@ public void OpenMissingSources()
             Assert.IsTrue(errors.HasErrors);
             Assert.IsNull(doc);
         }
+
+        [TestMethod]
+        public void TestJSONValueChanged()
+        {
+
+        string path1 = Path.Combine(Environment.CurrentDirectory, "JSON", "changed1.json");
+        string path2 = Path.Combine(Environment.CurrentDirectory, "JSON", "changed2.json");
+
+        ErrorContainer errorContainer = new ErrorContainer();
+
+            byte[] jsonString1 = File.ReadAllBytes(path1);
+            byte[] jsonString2 = File.ReadAllBytes(path2);
+
+            var jsonDictionary1 = MsAppTest.FlattenJson(jsonString1);
+            var jsonDictionary2 = MsAppTest.FlattenJson(jsonString2);
+
+            // IsMismatched on mismatched files
+            MsAppTest.CheckPropertyChangedRemoved(jsonDictionary1, jsonDictionary2, errorContainer, "");
+            MsAppTest.CheckPropertyAdded(jsonDictionary1, jsonDictionary2, errorContainer, "");
+
+            // Assume no mismatch
+            bool JSONPropertyAdded = false;
+            bool JSONPropertyRemoved = false;
+            bool JSONValueChanged = false;
+
+            // For every error received, check JSON Mismatches
+            foreach (var error in errorContainer)
+            {
+                if (error.Code == ErrorCode.JSONPropertyAdded)
```

Comment:
Do we have tests to validate that the message we output is correct? Like the nested paths are output in the correct way?

Reply:
Added!

Reply:
Made new changes we discussed!